operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java

```diff
@@ -0,0 +1,140 @@
+    Log.info("Creating RBAC into Namespace " + namespace);
+    List<HasMetadata> hasMetadata = k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + deploymentTarget + ".yml"))
+            .inNamespace(namespace).get();
+    hasMetadata.stream()
+            .map(b -> {
+              if ("Deployment".equalsIgnoreCase(b.getKind()) && b.getMetadata().getName().contains("operator")) {
+                ((Deployment) b).getSpec().getTemplate().getSpec().getContainers().get(0).setImagePullPolicy("Never");
+              }
+              return b;
+            }).forEach(c -> {
+              Log.info("processing part : " + c.getKind() + "--" + c.getMetadata().getName() + " -- " + namespace);
+              k8sclient.resource(c).inNamespace(namespace).createOrReplace();
+            });
+  }
+
+  private static void cleanRBACresourcesAndOperatorDeployment() throws FileNotFoundException {
+    Log.info("Deleting RBAC from Namespace " + namespace);
+    k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER +deploymentTarget+".yml"))
+            .inNamespace(namespace).delete();
+  }
+  private static void createCRD() throws FileNotFoundException {
+    Log.info("Creating CRD ");
+    k8sclient.load(new FileInputStream(TARGET_KUBERNETES_GENERATED_YML_FOLDER + "keycloaks.keycloak.org-v1.yml")).createOrReplace();
+  }
+
+  private static void registerReconcilers() {
+    Log.info("Registering reconcilers for operator : " + operator + " [" + operatorDeployment + "]");
+
+    for (Reconciler reconciler : reconcilers) {
+      final var config = configuration.getConfigurationFor(reconciler);
+      if (!config.isRegistrationDelayed()) {
+        Log.info("Register and apply : " + reconciler.getClass().getName());
+        OperatorProducer.applyCRDIfNeededAndRegister(operator, reconciler, configuration);
+      }
+    }
+  }
+
+  private static void createOperator() {
+    operator = new Operator(k8sclient, configuration);
+    operator.getConfigurationService().getClientConfiguration().setNamespace(namespace);
+  }
+
+  private static void createNamespace() {
+    Log.info("Creating Namespace " + namespace);
+    k8sclient.namespaces().create(new NamespaceBuilder().withNewMetadata().withName(namespace).endMetadata().build());
+  }
+
+  private static void calculateNamespace() {
+    namespace = "keycloak-test-" + UUID.randomUUID();
+  }
+
+  @AfterAll
+  public static void after() throws FileNotFoundException {
+
+    if (operatorDeployment == OperatorDeployment.local) {
+      Log.info("Stopping Operator");
+      operator.stop();
+
+      Log.info("Creating new K8s Client");
+      // create a new client bc operator has closed the old one
+      createK8sClient();
+    } else {
+      //cleanRBACresourcesAndOperatorDeployment();
+    }
+
+    Log.info("Deleting namespace : " + namespace);
+    //assertThat(k8sclient.namespaces().withName(namespace).delete()).isTrue();
```

Comment:
uncomment as well?

Reply:
If there are problems with `finalizers` cherry-pick this change:

https://github.com/keycloak/keycloak/pull/9849/files#diff-d3a84a1fd93698affd0fc0bc08c140043d55e02a5f5d2f913c73132eca4c8130R51

Reply:
:+1: